ravenml/utils/io_utils.py

```diff
@@ -0,0 +1,167 @@
+import os
+import shutil
+import boto3
+from pathlib import Path
+from queue import Queue
+from threading import Thread
+
```

Comment:
The implementations of these functions are from jigsaw, will be looking into replacing them to make them faster